### PR TITLE
Task/design

### PR DESCRIPTION
**Pull request by karstenlenz**
Please review my code. 
I implemented the design for my app. The functionality should remain the same.

task: https://github.com/karstenlenz/capstone-project/issues/17

### CODE REVIEW

File: src/Homepage/Homepage.js [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+import BulletList from '../common/BulletList'
 import Button from '../common/Button'
-import { Link } from 'react-router-dom'
+import FloatingButtonContainer from '../common/FloatingButtonContainer'
+import HeadlineUnderline from '../common/HeadlineUnderline'
 
 export default function () {
   return (
     <>
       <img alt="" src="/img/mirror.svg" />
```
Comment (JonasImm) [inline src/Homepage/Homepage.js:11]: Hier fehlt ein Alt Attribut

File: src/Motives/MotivesIntro.js [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
 import Button from '../common/Button'
-import { Link } from 'react-router-dom'
+import FloatingButtonContainer from '../common/FloatingButtonContainer'
+import HeadlineUnderline from '../common/HeadlineUnderline'
 
 export default function MotivesIntro() {
   return (
     <>
-      <h1>Schritt 1: Bedürfnisse</h1>
       <img alt="" src="/img/motives-intro.svg" />
```
Comment (JonasImm) [inline src/Motives/MotivesIntro.js:9]: Alt Attribut missing

File: src/Motives/__snapshots__/Motives.test.js.snap [hunk 3]
```diff
@@ -2,11 +2,24 @@
 
 exports[`The Motives component renders correctly 1`] = `
 <div>
-  <h1>
-    Bedürfnisse
-  </h1>
+  <img
+    alt=""
```
Comment (JonasImm) [inline src/Motives/__snapshots__/Motives.test.js.snap:6]: Alt Attribute missing

File: src/Questionnaire/Questionnaire.js [hunk 4]
```diff
@@ -27,14 +30,27 @@ export default function Questionnaire({ testIds }) {
 
   return (
     <>
-      <h1>Fragebogen {questionRound + 1} / 2</h1>
-      <p>"{personalityStyleData[currentTestIndex]?.name}"</p>
+      <img alt="" src="/img/questionnaire-intro.svg" />
```
Comment (JonasImm) [inline src/Questionnaire/Questionnaire.js:33]: Alt attribute missing

File: src/Questionnaire/Questionnaire.js [hunk 5]
```diff
@@ -27,14 +30,27 @@ export default function Questionnaire({ testIds }) {
...
         Frage {currentQuestionIndex + 1} / {questions.length}
       </h2>
       <h3>{questions[currentQuestionIndex]}</h3>
-      <button onClick={() => handleAnswer(false)}>(Eher) Nein</button>
-      <button onClick={() => handleAnswer(true)}>(Eher) Ja</button>
+
+      <ButtonRow>
+        <Button
+          btnType="white"
+          width="47.5"
```
Comment (JonasImm) [inline src/Questionnaire/Questionnaire.js:45]: Why do you define the width here and not in a styled component part?

File: src/Questionnaire/QuestionnaireIntro.js [hunk 6]
```diff
@@ -1,11 +1,14 @@
 import React from 'react'
 import Button from '../common/Button'
+import HeadlineUnderline from '../common/HeadlineUnderline'
 
 export default function QuestionnaireIntro({ onButtonClick }) {
   return (
     <>
-      <h1>Schritt 2: Fragebogen</h1>
       <img alt="" src="/img/questionnaire-intro.svg" />
```
Comment (JonasImm) [inline src/Questionnaire/QuestionnaireIntro.js:8]: Alt attribute missing

File: src/Result/PersonalityStyleInfo.js [hunk 7]
```diff
@@ -2,49 +2,64 @@ import React from 'react'
...
+            return <li key={tip}>{tip}</li>
+          })}
+        </BulletList>
+      </InfoBG>
       <h2>Weitere Informationen</h2>
-      <ul>
+      <BulletList listType="bullet_circle">
         {currentStyleData.furtherInfo.map((info, index) => {
           return (
             <li key={index}>
```
Comment (JonasImm) [inline src/Result/PersonalityStyleInfo.js:51]: I might marked this one before, but when mapping over an array, one should go for the ID as key values. Not the index position.

File: src/Result/Result.js [hunk 8]
```diff
@@ -1,63 +1,69 @@
...
+        <p>
+          Unser Testergebnis stellt nur eine Tendenz dar und ersetzt keine
+          psychologische Beratung. Wenn Sie sich unwohl fühlen, holen Sie sich
+          professionelle Hilfe.
+        </p>
+        <TextLink href="https://www.psychenet.de/de/hilfe-finden/schnelle-hilfe/krisenanlaufstellen.html">
+          Ansprechpartner &gt;
+        </TextLink>
+      </NoticeBox>
+      <img src="/img/style_info.svg" alt="" />
```
Comment (JonasImm) [inline src/Result/Result.js:24]: Alt Atribute missing

File: src/Result/Result.js [hunk 9]
```diff
@@ -1,63 +1,69 @@
...
+      <HeadlineUnderline>
+        <h1>Ergebnis</h1>
+      </HeadlineUnderline>
+      <ResultIntro>
+        Sie haben eben Tests für folgende Persönlichkeitsstile ausgefüllt:
+      </ResultIntro>
+      {resultData.map((result) => {
+        return (
+          <SectionBG key={result.id}>
```
Comment (JonasImm) [inline src/Result/Result.js:33]: Here you add the id, but above not?

File: src/Result/__snapshots__/Result.test.js.snap [hunk 10]
```diff
@@ -15,66 +18,82 @@ exports[`The Result component renders correctly 1`] = `
...
+      Ansprechpartner &gt;
     </a>
   </section>
-  <h1>
-    Ergebnis
-  </h1>
-  <p>
-    Sie haben eben Tests für folgende Persönlichkeitsstile ausgefüllt
+  <img
+    alt=""
```
Comment (JonasImm) [inline src/Result/__snapshots__/Result.test.js.snap:28]: Alt attribute missing

File: src/common/BulletList.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+const ListContainer = styled.ul`
+  padding: 0;
+  margin: 0 0 30px 0;
+
+  li {
+    list-style: none;
+    background: ${(props) =>
+      'url(/img/' + props.listType + '.svg) no-repeat left top;'};
+    padding: 0px 0px 3px 27px;
```
Comment (JonasImm) [inline src/common/BulletList.js:16]: Seems to be very accurate. Have you tried it on different screen sizes and mobile sizes?

File: src/common/BulletList.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    margin-bottom: 18px;
+  }
+
+  h3 {
+    margin: 0 0 0 0;
```
Comment (JonasImm) [inline src/common/BulletList.js:21]: Could be written in `margin: 0`

**Review by JonasImm — COMMENTED** [review]
Hey, nice styling so far. I`m impressed on your progress :) 
Just small things I found. One thing you seem to forget are the alt attributes on images. Or is that for a reason?
While opening your app on desktop mode, I don't have any hover effect with my mouse (e.g. a pointer instead of an cursor).  
Also, will there be any GO BACK button, in case I want to have a look on the last answer or go back to the intro section? 
GJ 💯

**Review by ninasander — APPROVED** [review]
Looks good to me! Maybe as a little suggestion, have you tried importing your svgs as React Components? If that works for your code, it might be easier to read than img links at first glance.

File: src/Homepage/Homepage.js [hunk 13]
```diff
@@ -1,13 +1,18 @@
 import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Homepage/Homepage.js:2]: You could also use `'styled-components/macro'`here, as you did in your global styles.

File: src/Homepage/Homepage.js (hunk 1 above)
Comment (macfrei) [reply src/Homepage/Homepage.js:11]: As long as the svg doesn't have an essential meaning to understand the page, there is no need to describe the image.

File: src/Motives/MotivesIntro.js (hunk 2 above)
Comment (macfrei) [reply src/Motives/MotivesIntro.js:9]: See comment above.

File: src/Motives/__snapshots__/Motives.test.js.snap (hunk 3 above)
Comment (macfrei) [reply src/Motives/__snapshots__/Motives.test.js.snap:6]: See comment above.

File: src/Questionnaire/Questionnaire.js [hunk 14]
```diff
@@ -1,6 +1,9 @@
 import React, { useEffect, useState } from 'react'
 import { useHistory } from 'react-router-dom'
 import { personalityStyleData } from '../data/personalityStyleData'
+import HeadlineUnderline from '../common/HeadlineUnderline'
+import Button from '../common/Button'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Questionnaire/Questionnaire.js:6]: Again, maybe with the `/macro`, I would say, that you either import it in all your components or not at all.

File: src/Questionnaire/Questionnaire.js (hunk 4 above)
Comment (macfrei) [reply src/Questionnaire/Questionnaire.js:33]: See comment above.

File: src/Questionnaire/QuestionnaireIntro.js (hunk 6 above)
Comment (macfrei) [reply src/Questionnaire/QuestionnaireIntro.js:8]: See comment above.

File: src/Result/PersonalityStyleInfo.js [hunk 15]
```diff
@@ -2,49 +2,64 @@ import React from 'react'
 import { useParams } from 'react-router-dom'
 import { personalityStyleData } from '../data/personalityStyleData'
 import { capitalizeFirstLetter } from '../common/util'
+import HeadlineUnderline from '../common/HeadlineUnderline'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Result/PersonalityStyleInfo.js:6]: Again with macro :)

File: src/Result/Result.js [hunk 16]
```diff
@@ -1,63 +1,69 @@
-import React, { useState } from 'react'
-import styled from 'styled-components'
+import React from 'react'
+import HeadlineUnderline from '../common/HeadlineUnderline'
+import NoticeBox from '../common/NoticeBox'
+import TextLink from '../common/TextLink'
+import { capitalizeFirstLetter } from '../common/util'
 import { personalityStyleData } from '../data/personalityStyleData'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/Result/Result.js:7]: Macro :)

File: src/Result/Result.js (hunk 8 above)
Comment (macfrei) [reply src/Result/Result.js:24]: See comment above.

File: src/Result/__snapshots__/Result.test.js.snap (hunk 10 above)
Comment (macfrei) [reply src/Result/__snapshots__/Result.test.js.snap:28]: See comment above.

File: src/common/FloatingButtonContainer.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/FloatingButtonContainer.js:1]: Macro :)

File: src/common/HeadlineUnderline.js [hunk 18]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/HeadlineUnderline.js:2]: Macro :D

File: src/common/HeadlineUnderline.js [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+import { ReactComponent as Underline } from '../img/underline.svg'
```
Comment (macfrei) [inline src/common/HeadlineUnderline.js:3]: Very nice!

File: src/common/InfoOverlay.js [hunk 20]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 import { useState } from 'react'
 import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/InfoOverlay.js:3]: Macro :)

File: src/common/NoticeBox.js [hunk 21]
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/NoticeBox.js:2]: Macro ;)

File: src/common/NoticeBox.js [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+
+export default function NoticeBox({ children }) {
+  const [isNoticeVisible, setIsNoticeVisible] = useState(true)
+
+  return (
+    <>
+      {isNoticeVisible && (
+        <SectionStyled>
+          <button onClick={() => setIsNoticeVisible(false)}>
+            <img alt="close overlay" src="/img/close.svg" />
```
Comment (macfrei) [inline src/common/NoticeBox.js:12]: Very good how you used the alt attribute here.

File: src/common/SectionBG.js [hunk 23]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/SectionBG.js:2]: Macro

File: src/common/TextLink.js [hunk 24]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import styled from 'styled-components'
```
Comment (macfrei) [inline src/common/TextLink.js:2]: Macro

**Review by macfrei — COMMENTED** [review]
I left a few comments, most of them about `styled-components/macro`. It looks really good! Keep up the good work!

**Comment by karstenlenz** [comment]
> Hey, nice styling so far. I`m impressed on your progress :)
> Just small things I found. One thing you seem to forget are the alt attributes on images. Or is that for a reason?
> While opening your app on desktop mode, I don't have any hover effect with my mouse (e.g. a pointer instead of an cursor).
> Also, will there be any GO BACK button, in case I want to have a look on the last answer or go back to the intro section?
> GJ 💯

Thanks for the review!
I left the alt attribute empty on purpose, so that they will be ignored by screenreaders. This is recommended behavior for images that are purely decorative and don't contain any important meaning.
see https://www.w3.org/WAI/tutorials/images/decorative/

I added the cursor, thanks for the hint.
I may add a back button for the result details page later. I didn't provide a back button in the test, because I didn't want users to cherry-pick the result they want.

File: src/Questionnaire/Questionnaire.js (hunk 5 above)
Comment (karstenlenz) [reply src/Questionnaire/Questionnaire.js:45]: When I change the width I also need to change "display" to "inline". I couldn't figure out how to do that with a styled component, so I put that logic into my Button component.

File: src/common/BulletList.js (hunk 11 above)
Comment (karstenlenz) [reply src/common/BulletList.js:16]: Yes, I tested it on different screen sizes. This controls the positioning of the bullet icon (circle or checkmark). I want a fixed distance on all screen sizes.

**Comment by carolinwagner** [comment]
I tested your App on an iPhone and it works perfectly :-) great job! 

Just a little note: On both the desktop and the iPhone version the background is still scrollable if you clicked on an "i"-Button at "Schritt 1: Bedürfnisse". Not sure if this is what you wanted.

**Comment by jennydely** [comment]
result of the QA test:
everything works good
at the moment it seems like there are just 2 tests results no matter, what i answer, is this right?

**Comment by karstenlenz** [comment]
> result of the QA test:
> everything works good
> at the moment it seems like there are just 2 tests results no matter, what i answer, is this right?

You should always fill out 2 questionnaires and thus get 2 results. That's intended. If you get 2 different personality styles depending on you answers it should be okay.

**Comment by karstenlenz** [comment]
> I tested your App on an iPhone and it works perfectly :-) great job!
> 
> Just a little note: On both the desktop and the iPhone version the background is still scrollable if you clicked on an "i"-Button at "Schritt 1: Bedürfnisse". Not sure if this is what you wanted.

Actually I'm not even sure myself if that's what I want. I put a new item in my backlog to improve the modal behavior and will work on it later.

**Pull request merged**